app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationService.kt

```diff
@@ -0,0 +1,791 @@
+//                    newNotifications.add(single)
+//                }
+//            }
+//        }
+
+        val notificationManagerCompat = NotificationManagerCompat.from(context)
+        // NOTE having multiple summary notifications: this here should still collapse them in only one occurrence
+        notificationManagerCompat.notify(newNotifications)
+    }
+
+    private fun createNotification(apiNotification: Notification, account: AccountEntity): NotificationWithIdAndTag? {
+        val baseNotification = createBaseNotification(apiNotification, account) ?: return null
+
+        return NotificationWithIdAndTag(
+            apiNotification.id,
+            account.id.toInt(),
+            baseNotification
+        )
+    }
+
+    // Only public for one test...
+    fun createBaseNotification(apiNotification: Notification, account: AccountEntity): android.app.Notification? {
+        val channelId = getChannelId(account, apiNotification.type) ?: return null
+
+        val body = apiNotification.rewriteToStatusTypeIfNeeded(account.accountId)
+
+        // Check for an existing notification matching this account and api notification
+        var existingAndroidNotification: android.app.Notification? = null
+        val activeNotifications = notificationManager.activeNotifications
+        for (androidNotification in activeNotifications) {
+            if (body.id == androidNotification.tag && account.id.toInt() == androidNotification.id) {
+                existingAndroidNotification = androidNotification.notification
+            }
+        }
+
+        notificationId++
+
+        val builder = if (existingAndroidNotification == null) {
+            getNotificationBuilder(body.type, account, channelId)
+        } else {
+            NotificationCompat.Builder(context, existingAndroidNotification)
+        }
+
+        builder
+            .setContentTitle(titleForType(body, account))
+            .setContentText(bodyForType(body, account.alwaysOpenSpoiler))
+
+        if (body.type == Notification.Type.MENTION || body.type == Notification.Type.POLL) {
+            builder.setStyle(
+                NotificationCompat.BigTextStyle()
+                    .bigText(bodyForType(body, account.alwaysOpenSpoiler))
+            )
+        }
+
+        var accountAvatar: Bitmap?
+        try {
+            val target = Glide.with(context)
+                .asBitmap()
+                .load(body.account.avatar)
+                .transform(RoundedCorners(20))
+                .submit()
+
+            accountAvatar = target.get()
+        } catch (e: ExecutionException) {
+            Log.d(TAG, "Error loading account avatar", e)
+            accountAvatar = BitmapFactory.decodeResource(context.resources, R.drawable.avatar_default)
+        } catch (e: InterruptedException) {
+            Log.d(TAG, "Error loading account avatar", e)
+            accountAvatar = BitmapFactory.decodeResource(context.resources, R.drawable.avatar_default)
+        }
```

Comment:
In Kotlin, `try` can return a value.

```suggestion
        val accountAvatar = try {
            Glide.with(context)
                .asBitmap()
                .load(body.account.avatar)
                .transform(RoundedCorners(20))
                .submit()
                .get()
        } catch (e: ExecutionException) {
            Log.d(TAG, "Error loading account avatar", e)
            BitmapFactory.decodeResource(context.resources, R.drawable.avatar_default)
        } catch (e: InterruptedException) {
            Log.d(TAG, "Error loading account avatar", e)
            BitmapFactory.decodeResource(context.resources, R.drawable.avatar_default)
        }
```